requirements.txt

```diff
@@ -1,5 +1,8 @@
 # this library requires the cx_Oracle package, but we source it from our
 # ready-made Oracle docker container: https://github.com/cityofaustin/atd-oracle-py
 arrow==0.17.*
-boto3==1.14.*
 knackpy==1.0.*
+numpy==1.21.*
```

Comment:
@Charlie-Henry, since we're not bringing numpy into the mix—can i press you to refactor this w/o pandas? For parsing the S3 json, [this line](https://github.com/cityofaustin/atd-finance-data/blob/7d05603d1dbdc4d69601e8b6e11a34246bd9af1b/s3_to_knack.py#L54) should work for you:

```python
    s3 = boto3.resource("s3")
    obj = s3.Object(bucket_name, fname)
    obj_data = obj.get()["Body"].read().decode()
    return json.loads(obj_data)
```

and for the replace transform:

```python
columns = {
    "TASK_ORDER_DEPT": "DEPT",
    "TASK_ORDER_ID": "TASK_ORDER",
    "TASK_ORDER_DESC": "NAME",
    "TASK_ORDER_STATUS": "Status",
    "TASK_ORDER_TYPE": "TK_TYPE",
    "TK_CURR_AMOUNT": "CURRENT_ESTIMATE",
    "CHARGED_AMOUNT": "CHARGEDAMOUNT",
    "TASK_ORDER_BAL": "BALANCE",
    "BYR_FDU": "BUYER_FDUS",
}


data = [
    {"TASK_ORDER_DEPT": 12345, "TASK_ORDER_ID": "Hello"}
]  # list of record dicts


replaced_data = [
    {dest_key: row.get(src_key) for src_key, dest_key in columns.items()}
    for row in data
]
```

To illustrate that list + dict comprehension. That transform is the same as

```python
replaced_data = []
for row in data:
    new_row = {}
    for src_key, dest_key in columns.items():
        new_row[dest_key] = row.get(src_key)
    replaced_data.append(new_row)
```